### ISSUE TITLE
CI: Mechanism to resolve dependencies for GnuPG (and other libraries?)

### ISSUE THREAD

**Issue by kriskwiatkowski** [rnpgp/rnp#611]
## Description

In #609 we started some discussion on how to resolve dependencies for GnuPG. Let's continue it here

The problem is that we currently hardcode versions of components needed to compile GnuPG. It would be good to have some nice and automatic way of doing this. 

Last idea was to parse this page https://gnupg.org/download/index.html.

(Please see #609 for details and trigger of the discussion)

**Comment by ronaldtse** [comment]
Thanks @flowher.

A shell script would probably be easiest here. I'm sure @erikbor doesn't mind whipping one up 😉

**Comment by ni4** [comment]
There is one more thing - currently AEAD may be checked only if gnupg/libgcrypt is built from the latest master, so I will disable it if corresponding version is not available. 
What about running Travis twice - with latest release and latest GnuPG master?

**Comment by flowher** [comment]
Good point. We can get "latest" version from gnupg git/svn (or whatever they are using).

**Comment by dd9jn** [comment]
FWIW, We have https://versions.gnupg.org/swdb.lst which is used by us to record the current versions of all components.  The script build-aux/getswdb.sh in the gnupg package is an interface to this.

**Comment by henrydcase** [comment]
Ah, this makes life easier. Thanks!

**Comment by dewyatt** [comment]
It looks like GnuPG and related items do not always have a buildable master branch (failure [here](https://travis-ci.org/riboseinc/rnp/jobs/352507161)), so we may have to do things a bit differently as far as our CI tests against GPG "beta".

**Comment by dd9jn** [comment]
On Mon, 12 Mar 2018 21:14, notifications@github.com said:
> It looks like GnuPG and related items do not always have a buildable master branch (failure [here](https://travis-ci.org/riboseinc/rnp/jobs/352507161)), so we may have to do things a bit differently as far as our CI tests against GPG "beta".

Sorry, I disabled a forthcoming API from libgpg-error but overlooked
that the constants are also internally.  Not sure why my pre-push checks
didn't caught it.

Thanks for noting.  Will be fixed soon.

The background is that we need to have a new libgpg-release but there is
not enough time to test and fully implement all intended new features.
Thus some need to be removed for the next release.


Salam-Shalom,

   Werner

-- 
#  Please read:  Daniel Ellsberg - The Doomsday Machine  #
Die Gedanken sind frei.  Ausnahmen regelt ein Bundesgesetz.

**Comment by ronaldtse** [comment]
@dd9jn thank you for the tip! What a surprise to see you reply here. Appreciate it! 👍
